processor/utils.ts

```diff
@@ -145,13 +150,28 @@ export const reformatHTML = (html: string, indent: number = 2): string => {
 
 export const toAttributes = (object: Record<string, any>, keys: string[] = []): MdxJsxAttribute[] => {
   let attributes: MdxJsxAttribute[] = [];
-  Object.entries(object).forEach(([name, value]) => {
-    if (keys.length > 0 && !keys.includes(name)) return;
+  Object.entries(object).forEach(([name, v]) => {
+    if ((keys.length > 0 && !keys.includes(name)) || typeof v === 'undefined') return;
+
+    let value: MdxJsxAttributeValueExpression | string;
+
+    if (typeof v === 'string') {
+      value = v;
+    } else {
+      const proxy = mdast(`{${v}}`);
+      const data = proxy.children[0].data as MdxJsxAttributeValueExpressionData;
```

Comment:
This `else` case is a bit hard to reason through without looking at the right test! Mind adding a comment here to clarify why/what the underlying issue is?

```suggestion
    } else {
      /* serialize <Image> components with JSX expressions in
       * their attributes (i.e. [border={false}]) back to MDX.
       */
      const proxy = mdast(`{${v}}`);
      const data = proxy.children[0].data as MdxJsxAttributeValueExpressionData;
```